### ISSUE TITLE
eos_config can't handle comma's when using templates

### ISSUE THREAD

**Issue by mchlrv** [ansible-collections/arista.eos#94]
##### SUMMARY
When trying to add snmp-location config via the eos_config module using "lines" in combination with "lookup" and "templates" it gives an error when the line contains comma's.

When using the same config command in "lines" as a list input there is no issue. When removing the comma's there is no issue too.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
eos_config

##### ANSIBLE VERSION
```
ansible 2.9.11
  config file = /Users/michael/ansible/ansible.cfg
  configured module search path = ['/Users/michael/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.11/libexec/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:48:26) [Clang 11.0.3 (clang-1103.0.32.62)]
```

##### CONFIGURATION

```
ANSIBLE_PIPELINING(/Users/michael/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/Users/michael/ansible/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=60s
RETRY_FILES_ENABLED(/Users/michael/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT
Arista vEOS device running version: 4.23.0.1F

##### STEPS TO REPRODUCE

```
---
- hosts: network
  gather_facts: no
  vars:
    snmp_location: 'Street 1, City, Country'

  tasks:
    - name: Check base configuration for changes
      eos_config:
        lines: "{{ lookup('template', 'snmp.j2') }}"
      when: ansible_network_os == 'eos'
```

snmp.j2
```
snmp-server location {{ snmp_location }}
```

##### EXPECTED RESULTS
Device gets configured

##### ACTUAL RESULTS
```
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/var/folders/c6/yrc04mj14tl128qfxv13xf980000gn/T/ansible_eos_config_payload_wSJg8A/ansible_eos_config_payload.zip/ansible/module_utils/network/eos/eos.py", line 174, in load_config
    response = conn.edit_config(commands, commit, replace)
  File "/var/folders/c6/yrc04mj14tl128qfxv13xf980000gn/T/ansible_eos_config_payload_wSJg8A/ansible_eos_config_payload.zip/ansible/module_utils/connection.py", line 185, in __rpc__
    raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
[WARNING]: Platform darwin on host arista.switch.tld is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python interpreter could change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
fatal: [arista.switch.tld]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "data": "City\r\n% Invalid input\r\narista.switch.tld(config-s-ansibl)#",
    "invocation": {
        "module_args": {
            "after": null,
            "auth_pass": null,
            "authorize": null,
            "backup": false,
            "backup_options": null,
            "before": null,
            "defaults": false,
            "diff_against": "session",
            "diff_ignore_lines": null,
            "host": null,
            "intended_config": null,
            "lines": [
                "snmp-server location Street 1",
                " City",
                " Country\n"
            ],
            "match": "line",
            "parents": null,
            "password": null,
            "port": null,
            "provider": null,
            "replace": "line",
            "running_config": null,
            "save_when": "never",
            "src": null,
            "ssh_keyfile": null,
            "timeout": null,
            "transport": null,
            "use_ssl": null,
            "username": null,
            "validate_certs": null
        }
    },
    "msg": "City\r\n% Invalid input\r\narista.switch.tld(config-s-ansibl)#"
}
```

```
ansible-playbook -i hosts network_config_test.yml -l arista.switch.tld -vvvvv
```

**Comment by GomathiselviS** [comment]
Hello @mchlrv , 

Thanks very much for your submission to Ansible.

**eos_config**'s line parameter expects a list. So can you please try the following

`
     arista.eos.eos_config:
        lines:
        - "{{ lookup('template', 'snmp.j2') }}"

`

**Comment by mchlrv** [comment]
Unfortunately I get the same error. I expected to fail because whenever I remove the comma's out of the variable value it works, even when not sending it as a list.

**Comment by GomathiselviS** [comment]
Following is my playbook and results. Please let me know if mine is different from your playbook. I do not see the issue.

```
---
- hosts: eos
  gather_facts: false
  vars:
    snmp_location: 'Street 1, City, Country'

  tasks:
    - name: Check base configuration for changes
      arista.eos.eos_config:
        lines:
        - "{{ lookup('template', 'snmp.j2') }}"
```



```
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying out features under development.
This is a rapidly changing source of code and can become unstable at any point.
ansible-playbook 2.11.0.dev0
  config file = /home/gosriniv/prjspace/my_collections/ansible.cfg
  configured module search path = ['/home/gosriniv/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/gosriniv/Project/ansible/lib/ansible
  ansible collection location = /home/gosriniv/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/gosriniv/Project/ansible/bin/ansible-playbook
  python version = 3.6.5 (default, Sep  4 2019, 12:23:33) [GCC 9.0.1 20190312 (Red Hat 9.0.1-0.10)]
Using /home/gosriniv/prjspace/my_collections/ansible.cfg as config file
host_list declined parsing /home/gosriniv/prjspace/my_collections/inventory_eos as it did not pass its verify_file() method
script declined parsing /home/gosriniv/prjspace/my_collections/inventory_eos as it did not pass its verify_file() method
auto declined parsing /home/gosriniv/prjspace/my_collections/inventory_eos as it did not pass its verify_file() method
toml declined parsing /home/gosriniv/prjspace/my_collections/inventory_eos as it did not pass its verify_file() method
[WARNING]:  * Failed to parse /home/gosriniv/prjspace/my_collections/inventory_eos with yaml plugin: We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 2 (char 1)  Syntax Error while loading YAML.   expected '<document start>', but found '<scalar>'  The error appears to be in
'/home/gosriniv/prjspace/my_collections/inventory_eos': line 2, column 1, but may be elsewhere in the file depending on the exact syntax problem.  The offending line appears to be:  [eos]
192.168.122.113 ansible_ssh_user=admin ansible_network_os=arista.eos.eos ansible_ssh_pass=admin ansible_user=admin ansible_password=admin ansible_become_pass=ansible ansible_become=True
ansible_become_method=ansible.netcommon.enable ansible_connection=ansible.netcommon.network_cli ^ here
  File "/home/gosriniv/Project/ansible/lib/ansible/inventory/manager.py", line 289, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/gosriniv/Project/ansible/lib/ansible/plugins/inventory/yaml.py", line 105, in parse
    raise AnsibleParserError(e)
[WARNING]:  * Failed to parse /home/gosriniv/prjspace/my_collections/inventory_eos with ini plugin: Invalid host pattern 'vars:' supplied, ending in ':' is not allowed, this character is
reserved to provide a port.
  File "/home/gosriniv/Project/ansible/lib/ansible/inventory/manager.py", line 289, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/gosriniv/Project/ansible/lib/ansible/plugins/inventory/ini.py", line 138, in parse
    raise AnsibleParserError(e)
[WARNING]: Unable to parse /home/gosriniv/prjspace/my_collections/inventory_eos as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: Collection arista.eos does not support Ansible version 2.11.0.dev0
redirecting (type: action) arista.eos.eos_config to arista.eos.eos

PLAYBOOK: eos_snmp.yml ***********************************************************************************************************************************************************************
1 plays in eos_snmp.yml

PLAY [eos] ***********************************************************************************************************************************************************************************
META: ran handlers
redirecting (type: action) arista.eos.eos_config to arista.eos.eos
[WARNING]: Collection ansible.netcommon does not support Ansible version 2.11.0.dev0

TASK [Check base configuration for changes] **************************************************************************************************************************************************
task path: /home/gosriniv/prjspace/my_collections/eos_snmp.yml:8
redirecting (type: action) arista.eos.eos_config to arista.eos.eos
redirecting (type: action) arista.eos.eos_config to arista.eos.eos
<192.168.122.113> ESTABLISH LOCAL CONNECTION FOR USER: gosriniv
<192.168.122.113> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f `"&& mkdir "` echo /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191 `" && echo ansible-tmp-1596573609.4520364-24723-262272772480191="` echo /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191 `" ) && sleep 0'
Using module file /home/gosriniv/prjspace/my_collections/collections/ansible_collections/arista/eos/plugins/modules/eos_config.py
<192.168.122.113> PUT /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/tmpvvv88qod TO /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191/AnsiballZ_eos_config.py
<192.168.122.113> EXEC /bin/sh -c 'chmod u+x /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191/ /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191/AnsiballZ_eos_config.py && sleep 0'
<192.168.122.113> EXEC /bin/sh -c '/usr/bin/env python /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191/AnsiballZ_eos_config.py && sleep 0'
<192.168.122.113> EXEC /bin/sh -c 'rm -f -r /home/gosriniv/.ansible/tmp/ansible-local-24603vj4a7u2f/ansible-tmp-1596573609.4520364-24723-262272772480191/ > /dev/null 2>&1 && sleep 0'
changed: [192.168.122.113] => {
    "changed": true,
    "commands": [
        "snmp-server location Street 1, City, Country"
    ],
    "diff": {
        "prepared": "--- system:/running-config\n+++ session:/ansible_1596573611-session-config\n@@ -11,6 +11,8 @@\n ip domain lookup vrf ansible source-interface Management1\n ip name-server vrf default 192.0.2.1\n ip name-server vrf default 192.0.2.2\n+!\n+snmp-server location Street 1, City, Country\n !\n spanning-tree mode mstp\n !"
    },
    "invocation": {
        "module_args": {
            "after": null,
            "backup": false,
            "backup_options": null,
            "before": null,
            "defaults": false,
            "diff_against": "session",
            "diff_ignore_lines": null,
            "intended_config": null,
            "lines": [
                "snmp-server location Street 1, City, Country\n"
            ],
            "match": "line",
            "parents": null,
            "provider": null,
            "replace": "line",
            "running_config": null,
            "save_when": "never",
            "src": null
        }
    },
    "session": "ansible_1596573611",
    "updates": [
        "snmp-server location Street 1, City, Country"
    ]
}
META: ran handlers
META: ran handlers

PLAY RECAP ***********************************************************************************************************************************************************************************
192.168.122.113            : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   



```

**Comment by GomathiselviS** [comment]
Following are my playbook and the run result. I do not see the issue. Please share your modified playbook.

```
---
- hosts: eos
  gather_facts: false
  vars:
    snmp_location: 'Street 1, City, Country'

  tasks:
    - name: Check base configuration for changes
      arista.eos.eos_config:
        lines:
        - "{{ lookup('template', 'snmp.j2') }}"

```

```

changed: [192.168.122.113] => {
    "changed": true,
    "commands": [
        "snmp-server location Street 1, City, Country"
    ],
    "diff": {
        "prepared": "--- system:/running-config\n+++ session:/ansible_1596574129-session-config\n@@ -11,6 +11,8 @@\n ip domain lookup vrf ansible source-interface Management1\n ip name-server vrf default 192.0.2.1\n ip name-server vrf default 192.0.2.2\n+!\n+snmp-server location Street 1, City, Country\n !\n spanning-tree mode mstp\n !"
    },
    "invocation": {
        "module_args": {
            "after": null,
            "backup": false,
            "backup_options": null,
            "before": null,
            "defaults": false,
            "diff_against": "session",
            "diff_ignore_lines": null,
            "intended_config": null,
            "lines": [
                "snmp-server location Street 1, City, Country\n"
            ],
            "match": "line",
            "parents": null,
            "provider": null,
            "replace": "line",
            "running_config": null,
            "save_when": "never",
            "src": null
        }
    },
    "session": "ansible_1596574129",
    "updates": [
        "snmp-server location Street 1, City, Country"
    ]
}
META: ran handlers
META: ran handlers

PLAY RECAP ***********************************************************************************************************************************************************************************
192.168.122.113            : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   


```

**Comment by mchlrv** [comment]
After going over this multiple times, it indeed turns out that lines expects a lists as you suggested. In my playbook I have two similar tasks. The first one only checks for changes the other one executes the changes. It turned out I only changed the lines list at the latter one, still causing the playbook to fail.

Since both tasks are very similar I didn't see it earlier. After stripping down the playbook to a bare minimum and run a couple of tests I figured out I was missing the list in the first task.

Thank you for you help!
